web/service/src/test/java/org/mobilitydata/gtfsvalidator/web/service/controller/RunValidatorEndpointTest.java

```diff
@@ -0,0 +1,128 @@
+  File mockFeedFile;
+  Path mockOutputPath;
+  File mockOutputPathToFile;
+  GoogleCloudPubsubMessage pubSubMessage;
+
+  @BeforeEach
+  public void setUp() throws Exception {
+    testJobId = "123";
+    jobMetaData = new JobMetadata(testJobId, "US");
+    mockFeedFile = mock(File.class);
+    mockOutputPath = mock(Path.class);
+    mockOutputPathToFile = mock(File.class);
+
+    var filePath = testJobId + "/feedFile.zip";
+    var msgText = "{\"name\": \"" + filePath + "\"}";
+    var innerMsg = new GoogleCloudPubsubMessage.Message();
+    innerMsg.setData(Base64.encodeBase64String(msgText.getBytes()));
+    pubSubMessage = new GoogleCloudPubsubMessage();
+    pubSubMessage.setMessage(innerMsg);
+
+    doReturn(jobMetaData).when(storageHelper).getJobMetadata(testJobId);
+    doReturn(mockOutputPath).when(storageHelper).getOutputPathForJob(testJobId);
+    doReturn(mockOutputPathToFile).when(mockOutputPath).toFile();
+  }
+
+  @Test
+  public void runValidatorSuccess() throws Exception {
+    doReturn(mockFeedFile)
+        .when(storageHelper)
+        .downloadFeedFileFromStorage(anyString(), anyString());
+
+    mockMvc
+        .perform(
+            MockMvcRequestBuilders.post("/run-validator")
+                .content(mapper.writeValueAsString(pubSubMessage))
+                .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(MockMvcResultMatchers.status().isOk());
+
+    // verify that the validationHandler is called with the downloaded feed file, output path, and
+    // country code
+    verify(validationHandler, times(1))
+        .validateFeed(mockFeedFile, mockOutputPath, jobMetaData.getCountryCode());
+
+    // verify that the validation output files are uploaded to storage
+    verify(storageHelper, times(1)).uploadFilesToStorage(testJobId, mockOutputPath);
+
+    // verify that the temp files and directory are deleted
+    verify(mockFeedFile, times(1)).delete();
+    verify(mockOutputPathToFile, times(1)).delete();
+  }
+
+  @Test
+  public void runValidatorStorageDownloadFailure() throws Exception {
+    doThrow(IOException.class)
+        .when(storageHelper)
+        .downloadFeedFileFromStorage(anyString(), anyString());
+
+    mockMvc
+        .perform(
+            MockMvcRequestBuilders.post("/run-validator")
+                .content(mapper.writeValueAsString(pubSubMessage))
+                .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(MockMvcResultMatchers.status().is5xxServerError());
+
+    // should not attempt validation
+    verify(validationHandler, times(0)).validateFeed(any(File.class), any(Path.class), anyString());
+
+    // should not upload to storage
+    verify(storageHelper, times(0)).uploadFilesToStorage(anyString(), any(Path.class));
+
+    // should not delete temp files
+    verify(mockFeedFile, times(0)).delete();
+    verify(mockOutputPathToFile, times(0)).delete();
+  }
+
+  @Test
+  public void runValidatorValidateFeedFailure() throws Exception {
```

Comment:
@davidgamez Hi David, I'm wondering if you can help me with this tests. I am trying to mock the validationHandler throwing an exception, but it does not seem to actually throw it using the same mechanism I used elsehwere. So this test "fails" because the handler is executing without throwing.

Reply:
@davidgamez once we work out this test, I think the PR is ready for review

Reply:
@briandonahue I committed a fix. It was missing the mocked file content as the `any` matcher doesn't match with null values. Some valuable explanation [here](https://github.com/mockito/mockito/issues/1188#issuecomment-333662952)

Reply:
@davidgamez Thank you!